src/d3a/models/strategy/area_agents/two_sided_revenue_fee.py

```diff
@@ -0,0 +1,96 @@
+
+
+def update_incoming_bid_with_fee(source_bid, original_bid, tax_percentage=None):
+    if source_bid is None:
+        return original_bid
+    return source_bid
+
+
+def update_incoming_offer_with_fee(source_offer_price, original_offer_price, tax_percentage):
+    if source_offer_price is None:
+        return original_offer_price * (1 + tax_percentage)
+    return source_offer_price + original_offer_price * tax_percentage
+
+
+# Protocol B
+def calculate_fee_revenue_from_clearing_trade(
+        bid_propagated, bid_original,
+        offer_propagated, offer_original,
+        trade_rate_source, tax_percentage_n
+):
+    demand_side_tax = 0 if bid_original == 0 else 1 - bid_propagated / bid_original
+    supply_side_tax = 0 if offer_original == 0 else offer_propagated / offer_original - 1
+    total_tax_percentage = demand_side_tax + supply_side_tax
+    revenue = trade_rate_source / (1 + total_tax_percentage)
+    fee_n = revenue * tax_percentage_n
+    trade_price = revenue + revenue * supply_side_tax
+    return revenue, fee_n, trade_price
+
+
+def calculate_fee_revenue_from_propagated_trade(
```

Comment:
This function doesnt seam to be used anywhere in the code.

Reply:
You are right, this is not needed here. Removing